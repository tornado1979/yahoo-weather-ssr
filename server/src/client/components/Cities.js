import React from 'react'
import propTypes from 'prop-types'

const Cities = ({ activeCity, cities }) => {
  return (
    <ul>
      {cities.map((city, idx) => <li className={activeCity === city ? 'active' : 'normal'} key={city + idx}>{city}</li>)}
    </ul>
  )
}

Cities.propTypes = {
  activeCity: propTypes.string.isRequired,
  cities: propTypes.arrayOf(propTypes.oneOfType([propTypes.string])).isRequired,
}

export default Cities
