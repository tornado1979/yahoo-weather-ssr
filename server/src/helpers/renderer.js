import React from 'react'
import { renderToString } from 'react-dom/server'
import { StaticRouter } from 'react-router-dom'
import { Provider } from 'react-redux'
import { renderRoutes } from 'react-router-config'
import serialize from 'serialize-javascript'

import Routes from '../client/Routes'

export default (req, store, context) => {
  const content = renderToString(
    <Provider store={store}>
      <StaticRouter context={context} location={req.path}>
        <div>{renderRoutes(Routes)}</div>
      </StaticRouter>
    </Provider>,
  )
  return `
  <!DOCTYPE html>
  <html>
    <head>
      <meta charset="UTF-8">
      <title>Server Side render application</title>
    </head>
   <body>
     <div id="root">${content}</div>
     <script>
       window.INITIAL_STATE = ${serialize(store.getState())}
     </script>
     <script src="bundle.js"></script>
   </body>
  </html>
  `
}
