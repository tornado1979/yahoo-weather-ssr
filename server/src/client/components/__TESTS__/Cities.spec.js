import React from 'react'
import renderer from 'react-test-renderer'

import Cities from '../Cities'

describe('Component Cities', () => {
  it('renders correctly', () => {
    const cities = ['Athens', 'Thessaloniki']
    const activeCity = 'Athens'

    const tree = renderer
      .create(<Cities activeCity={activeCity} cities={cities} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
