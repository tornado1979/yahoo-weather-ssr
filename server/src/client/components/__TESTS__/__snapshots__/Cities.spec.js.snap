// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Cities renders correctly 1`] = `
<ul>
  <li
    className="active"
  >
    Athens
  </li>
  <li
    className="normal"
  >
    Thessaloniki
  </li>
</ul>
`;
